### PR TITLE
Document ManagedOSVersion/ManagedOSVersionChannel

### LINKED ISSUE #299: Document ManagedOSVersion and ManagedOSVersionChannel
`ManagedOSVersion` and `ManagedOSVersionChannel` reference is missing from the documentation.

### PR DESCRIPTION
**Pull request by ldevulder**
Will fix #299.

### CODE REVIEW

File: versioned_docs/version-1.4/managedosversionchannel-reference.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+|------------------|--------|---------------|------------------------------------------------------------------------------|
+| options          | object | null          | Defines the optional informations that can be added in an OS channel         |
+| syncInterval     | string | empty         | Defines when to update the OS channel                                        |
+| type             | string | empty         | Defines the channel type, only `custom` is supported now                     |
+| upgradeContainer | object | null          | An upgrade container that can be defined. See [reference](#upgradecontainer) |
+
+#### upgradeContainer
+
+This allows to overwrite the default `upgrade` container executing the upgrade job on a node.
+The keys directly translate to the [container](https://kubernetes.io/docs/reference/kubernetes-api/workload-resources/pod-v1/#Container) specification.
```
Comment (davidcassany) [inline versioned_docs/version-1.4/managedosversionchannel-reference.md:41]: I don't think this is accurate enough, in fact `upgradeContainer` is of type `ContainerSpec` defined in system-upgrade-controller. Unless you verified system-upgrade-controller translates directly to the k8s pod resource (which is provably the case) I'd just mention this of type `ContainerSpec`, part of System Upgrade Controller project, and maps the the `ugprade` field of a System Upgrade Controller Plan. What about something in the spirit of:

`upgradeController` is mapped to a `v1.ContainerSpec` object (defined as part of [System Upgrade Controller](https://github.com/rancher/system-upgrade-controller)), which will be used as the `upgrade` field in an eventual System Upgrade Controller Plan created by the elemental-operator when upgrading to a `ManagedOSVersion` defined from this channel.

File: versioned_docs/version-1.4/managedosversion-reference.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+| Key          | Type   | Default value | Description                         |
+|--------------|--------|---------------|-------------------------------------|
+| displayName  | string | empty         | OS image name as seen in Rancher UI |
+| uri          | string | empty         | Fully qualified ISO image           |
+| upgradeImage | string | empty         | Fully qualified Containter image    |
+
+#### upgradeContainer
+
+This allows to overwrite the default `upgrade` container executing the upgrade job on a node.
+The keys directly translate to the [container](https://kubernetes.io/docs/reference/kubernetes-api/workload-resources/pod-v1/#Container) specification.
```
Comment (davidcassany) [inline versioned_docs/version-1.4/managedosversion-reference.md:81]: The same as for the managedosversionchannel.

File: docs/managedosversion-reference.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    upgradeImage: registry.suse.com/suse/sl-micro/6.0/baremetal-os-container:2.2.0
+  type: container
+  version: v2.2.0
+  ```
+  
+</details>
+
+#### metadata
+
+This describes the needed informations to define an OS image in Elemental.
```
Comment (davidcassany) [inline docs/managedosversion-reference.md:70]: Metadata can be of multiple types. The format of it is determined by the  `type` field.

`type: container` results into:

```
| Key          | Type   | Default value | Description                         |
|--------------|--------|---------------|-------------------------------------|
| displayName  | string | empty         | OS image name as seen in Rancher UI |
| upgradeImage | string | empty         | Fully qualified Containter image    |
```

and `type: iso` maps to:

```
| Key          | Type   | Default value | Description                                           |
|--------------|--------|---------------|-------------------------------------------------------|
| displayName  | string | empty         | OS image name as seen in Rancher UI                   |
| uri          | string | empty         | Fully qualified ISO image (OCI reference or HTTP URI) |
```

**Review by davidcassany — APPROVED** [review]
Just a couple of comments regarding the `upgradeContainer` and the  `metadata` of managedosversion types.

File: docs/managedosversion-reference.md (hunk 3 above)
Comment (ldevulder) [reply docs/managedosversion-reference.md:70]: Ok, I will split it, less clearer the way I wrote it.

File: docs/managedosversionchannel-reference.md [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+| options          | object | null          | Defines the optional informations that can be added in an OS channel         |
+| syncInterval     | string | empty         | Defines when to update the OS channel                                        |
+| type             | string | empty         | Defines the channel type, only `custom` is supported now                     |
+| upgradeContainer | object | null          | An upgrade container that can be defined. See [reference](#upgradecontainer) |
+
+#### upgradeContainer
+
+This allows to overwrite the default `upgrade` container executing the upgrade job on a node.
+
+`upgradeContainer` is mapped to a `v1.ContainerSpec` object (defined as part of [System Upgrade Controller](https://github.com/rancher/system-upgrade-controller)), which will be used as the upgrade field in an eventual System Upgrade Controller Plan created by the elemental-operator when upgrading to a `ManagedOSVersion` defined from this channel.
```
Comment (davidcassany) [inline docs/managedosversionchannel-reference.md:42]: What about something like:

```suggestion
This allows to overwrite the default `upgrade` field of System Upgrade Controller plans (see [upgrade compontents](/upgrade-lifecycle.md#components)) based on this ManagedOSVersion. These keys are translated by the System Upgrade Controller to a Kubernetes [container](https://kubernetes.io/docs/reference/kubernetes-api/workload-resources/pod-v1/#Container) specification. This is the container responsible of running an OS upgrade.
```

File: versioned_docs/version-1.2/managedosversion-reference.md [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+| Key          | Type   | Default value | Description                         |
+|--------------|--------|---------------|-------------------------------------|
+| displayName  | string | empty         | OS image name as seen in Rancher UI |
+| uri          | string | empty         | Fully qualified ISO image           |
+
+#### upgradeContainer
+
+This allows to overwrite the default `upgrade` container executing the upgrade job on a node.
+
+`upgradeContainer` is mapped to a `v1.ContainerSpec` object (defined as part of [System Upgrade Controller](https://github.com/rancher/system-upgrade-controller)), which will be used as the upgrade field in an eventual System Upgrade Controller Plan created by the elemental-operator when upgrading to a `ManagedOSVersion` defined from this channel.
```
Comment (davidcassany) [inline versioned_docs/version-1.2/managedosversion-reference.md:88]:
```suggestion
This allows to overwrite the default `upgrade` field of System Upgrade Controller plans (see 
[upgrade compontents](/upgrade-lifecycle.md#components)) based on this ManagedOSVersion. These keys  
are translated by the System Upgrade Controller to a Kubernetes [container](https://kubernetes.io/docs/reference/kubernetes-api/workload-resources/pod-v1/#Container)  
specification. This is the container responsible of running an OS upgrade.
```

File: versioned_docs/version-1.2/managedosversionchannel-reference.md [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+| options          | object | null          | Defines the optional informations that can be added in an OS channel         |
+| syncInterval     | string | empty         | Defines when to update the OS channel                                        |
+| type             | string | empty         | Defines the channel type, only `custom` is supported now                     |
+| upgradeContainer | object | null          | An upgrade container that can be defined. See [reference](#upgradecontainer) |
+
+#### upgradeContainer
+
+This allows to overwrite the default `upgrade` container executing the upgrade job on a node.
+
+`upgradeContainer` is mapped to a `v1.ContainerSpec` object (defined as part of [System Upgrade Controller](https://github.com/rancher/system-upgrade-controller)), which will be used as the upgrade field in an eventual System Upgrade Controller Plan created by the elemental-operator when upgrading to a `ManagedOSVersion` defined from this channel.
```
Comment (davidcassany) [inline versioned_docs/version-1.2/managedosversionchannel-reference.md:42]:
```suggestion
This allows to overwrite the default `upgrade` field of System Upgrade Controller plans (see 
[upgrade compontents](/upgrade-lifecycle.md#components)) based on any ManagedOSVersion from this ManagedOSVersionChannel. These keys  
are translated by the System Upgrade Controller to a Kubernetes [container](https://kubernetes.io/docs/reference/kubernetes-api/workload-resources/pod-v1/#Container)  
specification. This is the container responsible of running an OS upgrade.
```

File: versioned_docs/version-1.3/managedosversion-reference.md [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+| Key          | Type   | Default value | Description                         |
+|--------------|--------|---------------|-------------------------------------|
+| displayName  | string | empty         | OS image name as seen in Rancher UI |
+| uri          | string | empty         | Fully qualified ISO image           |
+
+#### upgradeContainer
+
+This allows to overwrite the default `upgrade` container executing the upgrade job on a node.
+
+`upgradeContainer` is mapped to a `v1.ContainerSpec` object (defined as part of [System Upgrade Controller](https://github.com/rancher/system-upgrade-controller)), which will be used as the upgrade field in an eventual System Upgrade Controller Plan created by the elemental-operator when upgrading to a `ManagedOSVersion` defined from this channel.
```
Comment (davidcassany) [inline versioned_docs/version-1.3/managedosversion-reference.md:88]:
```suggestion
This allows to overwrite the default `upgrade` field of System Upgrade Controller plans (see 
[upgrade compontents](/upgrade-lifecycle.md#components)) based on this ManagedOSVersion. These keys  
are translated by the System Upgrade Controller to a Kubernetes [container](https://kubernetes.io/docs/reference/kubernetes-api/workload-resources/pod-v1/#Container)  
specification. This is the container responsible of running an OS upgrade.
```

File: versioned_docs/version-1.3/managedosversionchannel-reference.md [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+| options          | object | null          | Defines the optional informations that can be added in an OS channel         |
+| syncInterval     | string | empty         | Defines when to update the OS channel                                        |
+| type             | string | empty         | Defines the channel type, only `custom` is supported now                     |
+| upgradeContainer | object | null          | An upgrade container that can be defined. See [reference](#upgradecontainer) |
+
+#### upgradeContainer
+
+This allows to overwrite the default `upgrade` container executing the upgrade job on a node.
+
+`upgradeContainer` is mapped to a `v1.ContainerSpec` object (defined as part of [System Upgrade Controller](https://github.com/rancher/system-upgrade-controller)), which will be used as the upgrade field in an eventual System Upgrade Controller Plan created by the elemental-operator when upgrading to a `ManagedOSVersion` defined from this channel.
```
Comment (davidcassany) [inline versioned_docs/version-1.3/managedosversionchannel-reference.md:42]:
```suggestion
This allows to overwrite the default `upgrade` field of System Upgrade Controller plans (see 
[upgrade compontents](/upgrade-lifecycle.md#components)) based on any ManagedOSVersion from this ManagedOSVersionChannel. These keys  
are translated by the System Upgrade Controller to a Kubernetes [container](https://kubernetes.io/docs/reference/kubernetes-api/workload-resources/pod-v1/#Container)  
specification. This is the container responsible of running an OS upgrade.
```

File: versioned_docs/version-1.4/managedosversionchannel-reference.md [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+| options          | object | null          | Defines the optional informations that can be added in an OS channel         |
+| syncInterval     | string | empty         | Defines when to update the OS channel                                        |
+| type             | string | empty         | Defines the channel type, only `custom` is supported now                     |
+| upgradeContainer | object | null          | An upgrade container that can be defined. See [reference](#upgradecontainer) |
+
+#### upgradeContainer
+
+This allows to overwrite the default `upgrade` container executing the upgrade job on a node.
+
+`upgradeContainer` is mapped to a `v1.ContainerSpec` object (defined as part of [System Upgrade Controller](https://github.com/rancher/system-upgrade-controller)), which will be used as the upgrade field in an eventual System Upgrade Controller Plan created by the elemental-operator when upgrading to a `ManagedOSVersion` defined from this channel.
```
Comment (davidcassany) [inline versioned_docs/version-1.4/managedosversionchannel-reference.md:42]:
```suggestion
This allows to overwrite the default `upgrade` field of System Upgrade Controller plans (see 
[upgrade compontents](/upgrade-lifecycle.md#components)) based on any ManagedOSVersion from this ManagedOSVersionChannel. These keys  
are translated by the System Upgrade Controller to a Kubernetes [container](https://kubernetes.io/docs/reference/kubernetes-api/workload-resources/pod-v1/#Container)  
specification. This is the container responsible of running an OS upgrade.
```

**Pull request merged**